tests/Unit/RuleSet/DeclarationBlockTest.php

```diff
@@ -48,4 +52,78 @@ public function implementsPositionable(): void
     {
         self::assertInstanceOf(Positionable::class, $this->subject);
     }
+
+    /**
+     * @return array<non-empty-string, array{0: non-empty-string}>
+     */
+    public static function provideSelector(): array
+    {
+        return [
+            'type' => ['body'],
+            'class' => ['.teapot'],
+            'type & class' => ['img.teapot'],
+            'id' => ['#my-mug'],
+            'type & id' => ['h2#my-mug'],
+            'pseudo-class' => [':hover'],
+            'type & pseudo-class' => ['a:hover'],
+            '`not`' => [':not(#your-mug)'],
+            'pseudo-element' => ['::before'],
+            'attribute with `"`' => ['[alt="{}()[]\\"\'"]'],
+            'attribute with `\'`' => ['[alt=\'{}()[]"\\\'\']'],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param non-empty-string $selector
+     *
+     * @dataProvider provideSelector
+     */
+    public function parsesSingleSelector(string $selector): void
+    {
+        $subject = DeclarationBlock::parse(new ParserState($selector . '{}', Settings::create()));
+
+        self::assertNotNull($subject);
```

Comment:
Please let's be as specific as possible in the tests using `assertInstanceOf` instead (also in the other test).